notify/impl.go

```diff
@@ -789,17 +798,38 @@ func (n *Wechat) Notify(ctx context.Context, as ...*types.Alert) (bool, error) {
 		n.accessTokenAt = time.Now()
 	}
 
-	msg := &weChatMessage{
-		Text: weChatMessageContent{
+	var msg *weChatMessage
+	if n.conf.ToAppChat != "" {
+		msg = &weChatMessage{
+			ToAppChat: tmpl(n.conf.ToAppChat),
+			AgentID:   tmpl(n.conf.AgentID),
+			Safe:      "0",
+		}
+	} else {
+		msg = &weChatMessage{
+			ToUser:  tmpl(n.conf.ToUser),
+			ToParty: tmpl(n.conf.ToParty),
+			Totag:   tmpl(n.conf.ToTag),
+			AgentID: tmpl(n.conf.AgentID),
+			Safe:    "0",
+		}
+	}
+	msg.Type = n.conf.MessageType
+	if msg.Type == "textcard" {
+		msg.TextCard = weChatAppMessageContent{
+			Title:       tmpl(n.conf.Title),
+			URL:         tmpl(n.conf.URL),
+			Description: tmpl(n.conf.Description),
+			Btntxt:      tmpl(n.conf.Btntxt),
+		}
+	} else {
+		msg.Text = weChatMessageContent{
 			Content: tmpl(n.conf.Message),
-		},
-		ToUser:  tmpl(n.conf.ToUser),
-		ToParty: tmpl(n.conf.ToParty),
-		Totag:   tmpl(n.conf.ToTag),
-		AgentID: tmpl(n.conf.AgentID),
-		Type:    "text",
-		Safe:    "0",
+		}
 	}
+
+	s, _ := json.Marshal(msg)
+	level.Debug(n.logger).Log("msg", "wechat Request", "body", s)
```

Comment:
We don't log JSON payload usually as it can be very verbose. And it duplicates the encoding already done at L838.

Reply:
May this also contain secrets?

Reply:
> May this also contain secrets?

Json payload only have post data, no secrets, the secrets used to get accesstoken from wechat.

Reply:
> We don't log JSON payload usually as it can be very verbose. And it duplicates the encoding already done at L838.

OK I delete it